hermes/probe/metrics/metrics.go

```diff
@@ -0,0 +1,183 @@
+	UNKNOWN_FILE_FOUND
+	// ALL_FILES_MISSING indicates that all of the Hermes files were missing.
+	ALL_FILES_MISSING
+)
+
+var (
+	// ProbeOps maps ProbeOperation constants to their metric label string equivalent.
+	ProbeOps = map[ProbeOperation]string{
+		TOTAL_PROBE_RUN: "total_probe_run",
+		CHECK_NIL:       "check_nil",
+		READ_AND_VERIFY: "read_and_verify",
+		DELETE_FILE:     "delete_file",
+		CREATE_FILE:     "create_file",
+	}
+	// ApiCalls maps ApiCall constants to their metric label string equivalent.
+	ApiCalls = map[ApiCall]string{
+		API_LIST_FILES:  "list_files",
+		API_CREATE_FILE: "create_file",
+		API_DELETE_FILE: "delete_file",
+		API_GET_FILE:    "get_file",
+	}
+	// ExitStatuses maps ExitStatus constants to their metric label string equivalent.
+	ExitStatuses = map[ExitStatus]string{
+		SUCCESS:                "success",
+		OP_TIMEOUT:             "op_timeout",
+		PROBE_FAILED:           "probe_failed",
+		API_CALL_FAILED:        "api_call_failed",
+		FILE_MISSING:           "file_missing",
+		BUCKET_MISSING:         "bucket_missing",
+		FILE_CORRUPTED:         "file_corrupted",
+		FILE_READ_FAILURE:      "file_read_failure",
+		FILE_METADATA_MISMATCH: "file_metadata_mismatch",
+		UNKNOWN_FILE_FOUND:     "unknown_file_found",
+		ALL_FILES_MISSING:      "all_files_missing",
+	}
+)
+
+// Metrics stores the cumulative metrics for probe runs for a target.
+type Metrics struct {
+	// probeOpLatency is used to record latency values with distinct labels
+	// per exit status per probe operation per target.
+	// Recommended usage: probeOpLatency[ProbeOperation][ExitStatus].Metric("latency").AddFloat64(<val>)
+	ProbeOpLatency map[string]map[string]*metrics.EventMetrics
+	// apiCallLatency is used to record latency values with distinct labels
+	// per exit status per API call per target.
+	// Recommended usage: apiCallLatency[ApiCall][ExitStatus].Metric("latency").AddFloat64(<val>)
+	ApiCallLatency map[string]map[string]*metrics.EventMetrics
+}
+
+// NewMetrics creates a new *Metrics object and initialises the fields inside it.
+// Arguments:
+//	- conf: pass a HermesProbeDef config
+//	- target: pass the target for which metrics are to be collected.
+// Returns:
+//	- m: returns an initialised *Metrics object.
+//	- err: returns an error if a latency distribution cannot be created from the config proto.
+func NewMetrics(conf *probepb.HermesProbeDef, target *probepb.Target) (*Metrics, error) {
+	m := &Metrics{
+		ProbeOpLatency: make(map[string]map[string]*metrics.EventMetrics, len(ProbeOps)),
+		ApiCallLatency: make(map[string]map[string]*metrics.EventMetrics, len(ApiCalls)),
+	}
+
+	probeOpLatDist, err := metrics.NewDistributionFromProto(conf.GetProbeLatencyDistribution())
+	if err != nil {
+		return nil, fmt.Errorf("invalid argument: error creating probe latency distribution from the specification (%v): %w", conf.GetProbeLatencyDistribution(), err)
+	}
+
+	for _, op := range ProbeOps {
```

Comment:
If you switch to using the numeric identifier in the map, this will change to
```
for op := range ProbeOps {
```

Reply:
Ack.